torch/testing/_internal/opinfo/definitions/nested.py

```diff
@@ -0,0 +1,277 @@
+):
+    if not op_kwargs:
+        op_kwargs = {}
+
+    for njt in _sample_njts(
+        device=device, dtype=dtype, requires_grad=requires_grad, dims=[2, 3, 4]
+    ):
+        # dim-wise reduction; includes reduction over the ragged dim
+        # NB: reduction over the batch dim is not supported!
+        # TODO: Cover this in the set of error inputs
+        for dim in range(1, njt.dim()):
+            for keepdim in [False, True]:
+                yield SampleInput(
+                    njt, kwargs={**op_kwargs, "dim": dim, "keepdim": keepdim}
+                )
+
+        # full reduction
+        yield SampleInput(njt, kwargs=dict(op_kwargs))
+
+
+def unsupported_sample_inputs_func(op_name):
+    def _f(op_info, device, dtype, requires_grad, op_name=op_name, **kwargs):
+        raise RuntimeError(
+            f"OpInfo for {op_name} does not support NJT. Support can be added by modifying "
+            "torch/testing/_internal/opinfo/definitions/nested.py."
+        )
+
+    return _f
+
+
+def unsupported_reference(op_name):
+    def _f(op, sample):
+        raise RuntimeError(
+            f"OpInfo for {op_name} does not define a ref() function. Support can be added by "
+            "modifying torch/testing/_internal/opinfo/definitions/nested.py."
+        )
+
+    return _f
+
+
+# === BEGIN OP-SPECIFIC SAMPLE INPUTS FUNCS ===
+def sample_inputs_mvl_gamma(p):
+    return partial(sample_inputs_elementwise_njt_unary, op_kwargs={"p": p})
+
+
+def sample_inputs_polygamma_n(n):
+    return partial(sample_inputs_elementwise_njt_unary, op_kwargs={"n": n})
+
+
+def sample_inputs_special_polygamma_n(n):
+    return partial(sample_inputs_elementwise_njt_unary, op_kwargs={"n": n})
+
+
+sample_inputs_nn_functional_threshold = partial(
+    sample_inputs_elementwise_njt_unary,
+    op_kwargs={"threshold": float.fromhex("0x1.3ap-3"), "value": -9},
+)
+# === END OP-SPECIFIC SAMPLE INPUTS FUNCS ===
+
+
+# Mapping of OpInfo full names -> sample_inputs_funcs, which define the set of sample inputs
+# (involving NJTs) to pass to the op. Full name consists of the OpInfo's name and variant name
+# separated by a period (e.g. special.polygamma.special_polygamma_n_0). These are necessary
+# to specify if they cannot be auto-generated for some reason. Try to keep these sorted
+# in alphabetical order!
+njt_sample_inputs = {
+    **{f"mvlgamma.mvlgamma_p_{p}": sample_inputs_mvl_gamma(p=1) for p in (1, 3, 5)},
+    "nn.functional.threshold": sample_inputs_nn_functional_threshold,
+    **{f"polygamma.polygamma_n_{n}": sample_inputs_polygamma_n(n=n) for n in range(5)},
+    "special.polygamma.special_polygamma_n_0": sample_inputs_special_polygamma_n(n=0),
+}
+
+
+# Translates an OpInfo entry to one that operates on NJTs.
+def translate_opinfo(op):
+    new_op = copy(op)
+    new_op.supports_njt = True
+
+    if op.full_name in njt_sample_inputs:
+        new_op.sample_inputs_func = njt_sample_inputs[op.full_name]
```

Comment:
One reason to have a separate `nt_sample_inputs_func` rather than reuse the original `sample_inputs_func` field is if the person writing the sample inputs wants to do some transformation on the original inputs to produce the nested ones. 

But I'm also sure how practical automated transforms are on the original inputs compared to just writing test cases manually:
- For the simple case, where the nested tensor logic is unaffected by the non-tensor args, the bit of extra coverage that gives is that we'd be testing that the non-tensor args are properly propagated to the desugared call.  In this case, manually writing a few inputs to spot-check should be enough.
- For the case where nested tensor logic must take into account the argument, e.g. "dim" or "dims" its possible that at a certain point, it may just be more practical to manually write the sample inputs - e.g. for view ops.

Reply:
> One reason to have a separate nt_sample_inputs_func rather than reuse the original sample_inputs_func field is if the person writing the sample inputs wants to do some transformation on the original inputs to produce the nested ones.

I think this would still be possible with the current structure - you'd basically write a transformation func that utilizes the original `sample_inputs_func` to produce the new one. IIUC it's not strictly necessary to have a separate `nt_sample_inputs_func` field to accomplish this.

> But I'm also not sure how practical automated transforms are on the original inputs compared to just writing test cases manually:
> For the simple case, where the nested tensor logic is unaffected by the non-tensor args, the bit of extra coverage that gives is that we'd be testing that the non-tensor args are properly propagated to the desugared call. In this case, manually writing a few inputs to spot-check should be enough.

True, and maybe it's worth revisiting because it's nice that we can generally expect existing `OpInfo`s to cover non-tensor args pretty extensively. I don't think anything precludes us from adding this sorta thing on in the future.

> For the case where nested tensor logic must take into account the argument, e.g. "dim" or "dims" its possible that at a certain point, it may just be more practical to manually write the sample inputs - e.g. for view ops.
 
I'm glad you bring up view ops - I've been thinking of some view-specific testing we could do (it's certainly useful to have different views as input to a torch.compiled function to test out view fake-ification; we could move those tests over from `test/dynamo/test_subclasses.py`). I do agree that it's probably more practical to manually write the view sample inputs to ensure we're covering everything we want to, but there's still some view-specific testing we could write generally. There's always a tradeoff between generality and exhaustive coverage, and I think we want a little of both general and specific tests here.

Reply:
> I think this would still be possible with the current structure - you'd basically write a transformation func that utilizes the original sample_inputs_func to produce the new one. IIUC it's not strictly necessary to have a separate nt_sample_inputs_func field to accomplish this.

Good point, though I can imagine it being easier to write sample inputs function that operate generically over a set of operators. 

> There's always a tradeoff between generality and exhaustive coverage, and I think we want a little of both general and specific tests here.

Agreed. Definitely good to have a good baseline of testing through the OpInfo in addition to specific tests.